packages/ui-library/src/components/token-amount-usd-input/index.tsx

```diff
@@ -0,0 +1,346 @@
+  onBlur: () => void;
+}
+
+const calculateUsdAmount = ({
+  value,
+  token,
+  tokenPrice,
+}: {
+  value?: string;
+  token?: Nullable<Token>;
+  tokenPrice?: bigint;
+}) =>
+  isUndefined(value) || value === '' || isUndefined(tokenPrice) || !token
+    ? '0'
+    : parseFloat(formatUnits(parseUnits(value, token.decimals) * tokenPrice, token.decimals + 18)).toFixed(2);
+
+const calculateTokenAmount = ({ value, tokenPrice }: { value?: string; tokenPrice?: bigint }) =>
+  isUndefined(value) || value === '' || isUndefined(tokenPrice)
+    ? '0'
+    : formatUnits(parseUnits(value, 18 * 2) / tokenPrice, 18).toString();
+
+const validator = ({
+  nextValue,
+  decimals,
+  onChange,
+}: {
+  nextValue: string;
+  onChange: (newValue?: string) => void;
+  decimals: number;
+}) => {
+  const newNextValue = nextValue.replace(/,/g, '.');
+  // sanitize value
+  const inputRegex = RegExp(`^\\d*(?:\\\\[.])?\\d{0,${decimals}}$`);
+
+  if (inputRegex.test(newNextValue.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'))) {
+    onChange(newNextValue.startsWith('.') ? `0${newNextValue}` : newNextValue || undefined);
+  }
+};
+
+const TokenInput = ({ onChange, value, token, tokenPrice, onBlur, onFocus, disabled }: InputProps) => {
+  const {
+    palette: { mode },
+  } = useTheme();
+  const usdAmount = calculateUsdAmount({ value, token, tokenPrice });
+
+  return (
+    <ContainerBox flexDirection="column">
+      <FormControl variant="standard" fullWidth>
+        <Input
+          id="component-simple"
+          onChange={(evt) =>
+            validator({
+              onChange,
+              nextValue: evt.target.value,
+              decimals: token?.decimals || 18,
+            })
+          }
+          value={value || ''}
+          onFocus={onFocus}
+          onBlur={onBlur}
+          placeholder="0"
+          disableUnderline
+          inputProps={{ style: { color: getInputColor({ disabled, mode, hasValue: !isUndefined(value) }) } }}
+          sx={{ ...buildTypographyVariant(mode).h6, fontWeight: '700', color: 'inherit' }}
+        />
+      </FormControl>
+      <Typography variant="bodySmall" color={getSubInputColor({ mode, hasValue: !isUndefined(value) })}>
+        ≈{` $${usdAmount}`}
+      </Typography>
+    </ContainerBox>
+  );
+};
+
+// TODO
+const UsdInput = ({ onChange, value, token, tokenPrice, onBlur, onFocus, disabled }: InputProps) => {
+  const {
+    palette: { mode },
+  } = useTheme();
+  const tokenAmount = calculateTokenAmount({ value, tokenPrice });
+
+  return (
+    <ContainerBox flexDirection="column">
+      <FormControl variant="standard">
+        <Input
+          id="component-simple"
+          onChange={(evt) =>
+            validator({
+              onChange,
+              nextValue: evt.target.value,
+              decimals: 2,
+            })
+          }
+          value={value || ''}
+          onFocus={onFocus}
+          onBlur={onBlur}
+          startAdornment="$"
+          placeholder="0"
```

Comment:
![Screenshot 2024-02-14 at 09 10 13](https://github.com/Mean-Finance/dca-fe/assets/144699868/44cfe6c4-d323-4c36-9797-00620a29b912)

I found that autoComplete is making the input look nasty